lib/max_subarray.rb

```diff
@@ -1,8 +1,16 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) where n is the value of nums
+# Space Complexity: O(1)
 def max_sub_array(nums)
-    return 0 if nums == nil
-    
-    raise NotImplementedError, "Method not implemented yet!"
+  return 0 if nums == nil
+
+  max_so_far = nil
```

Comment:
Why not just start `max_so_far` to the smallest integer possible, to simplify the if statement.